pkg/indexers/apiexport.go

```diff
@@ -30,25 +30,21 @@ const (
 	APIExportByIdentity = "APIExportByIdentity"
 	// APIExportBySecret is the indexer name for retrieving APIExports by secret.
 	APIExportBySecret = "APIExportSecret"
+	// APIExportByClaimedIdentities is the indexer name for retrieving APIExports that have a permission claim for a
+	// particular identity hash.
+	APIExportByClaimedIdentities = "APIExportByClaimedIdentities"
 )
 
 // IndexAPIExportByIdentity is an index function that indexes an APIExport by its identity hash.
 func IndexAPIExportByIdentity(obj interface{}) ([]string, error) {
-	apiExport, ok := obj.(*apisv1alpha1.APIExport)
-	if !ok {
-		return []string{}, fmt.Errorf("obj %T is not an APIExport", obj)
-	}
-
```

Comment:
Noticed we have these type checks present across all indexers -- are we considering dropping them everywhere?

Reply:
Yes

Reply:
Could we also strictly type this stuff? Why `obj interface{}` ?

Reply:
Index function